src/controllers/notifications.js

```diff
@@ -71,6 +73,16 @@ const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 exports.post = function * postNotification () {
   const notification = yield requestUtil.validateBody(this, 'Notification')
   log.debug('Got notification: ' + JSON.stringify(notification))
+
+  const verifySignature = this.config.get('notifications.must_verify')
+  if (verifySignature) {
+    const result = model.verifySignature(notification, this.config)
+    if (!result.valid) {
+      log.warn('Signature verification failed: ' + result.error)
+      throw new UnprocessableEntityError('Notification failed signature verification')
```

Comment:
> When it happens, we wouldn't want to flood the servers. 
> How about logging an error and returning 200, like we do here?

There's no way to know anything is failing unless there's a separate process parsing the logs. I prefer explicitly returning an error code and letting the caller know that something is wrong.